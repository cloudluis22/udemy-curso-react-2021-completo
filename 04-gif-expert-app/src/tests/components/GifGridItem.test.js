import React from 'react';;
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en el compontente de "GifGridItem"', () => {

    const title = 'Titulo'
    const url = 'https://urlejemplo.com.mx';
    const wrapper = shallow(<GifGridItem
        title={title}
        url={url}
    />);

    test('Debe mostrar el componente correctamente.', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de contener un parrafo con el texto.', () => {
        expect(wrapper.text().trim()).toBe(title);
    })

    test('La imagen debe contener los datos correctos de el tÃ­tulo y el URL del gif.', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('El componente debe de tener un div con la clase "animate__fadeIn"', () => {

        const className = 'animate__fadeIn'

        const div = wrapper.find('div');
        expect(div.hasClass(className)).toBe(true);

    })


})
