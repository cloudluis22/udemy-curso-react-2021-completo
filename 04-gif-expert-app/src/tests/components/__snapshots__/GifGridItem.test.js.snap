// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el compontente de "GifGridItem" Debe mostrar el componente correctamente. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://urlejemplo.com.mx"
  />
  <p>
    Titulo
  </p>
</div>
`;
